enhancements/machine-api/aws-imds-v2-support.md

```diff
@@ -0,0 +1,260 @@
+---
+title: aws-imdsv2-support
+authors:
+  - "@lobziik"
+reviewers:
+  - "@patrickdillon"
+  - "@JoelSpeed"
+approvers:
+  - "@patrickdillon"
+  - "@JoelSpeed"
+creation-date: 2022-03-28
+last-updated: 2022-03-28
+tracking-link: 
+  - [OCPCLOUD-1436](https://issues.redhat.com/browse/OCPCLOUD-1436)
+---
+
+# Support AWS IMDSv2 for EC2 machines
+
+## Summary
+
+For increasing security and add protection against open firewalls, reverse proxies, and SSRF vulnerabilities Amazon introduced
+[enhancements to the EC2 Instance Metadata Service](http://aws.amazon.com/blogs/security/defense-in-depth-open-firewalls-reverse-proxies-ssrf-vulnerabilities-ec2-instance-metadata-service/)
+(IMDSv2). It is a session-based interaction method with Instance Metadata Service (IMDS), in other words,
+all requests to `http://169.254.169.254` will require prior PUT requests for obtaining token and such token must be used in further requests.
+This document describes and proposes several interfaces for configuring IMDS behaviour for newly created machines via Machine API and
+during the installation procedure.
+
+## Motivation
+
+By adding mechanisms for configuring IMDS behaviour we will allow our end-users to enhance the security of their clusters and simplify Amazon [recommended best practices](https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-standards-fsbp-controls.html#fsbp-ec2-8) implementation.
+At this point, IMDSv2 enablement might be done manually via AWS console, but providing such settings on Machine API is a quite loud request from our end-users.
+
+### Goals
+
+- provide mechanisms to enable/enforce IMDSv2 for newly created machines via Machine API
+- provide mechanisms to enable/enforce IMDSv2 for master machines during installation procedure for newly created clusters
+
+### Non-Goals
+
+- perform automated migration between using IMDSv1 and IMDSv2 for existing infrastructure
+
+## Proposal
+
+ - Add new options into install-config for Othe CP installation program which will be passed as metadata service settings during virtual machines creation
```

Comment:
`Othe` seems to be a typo. What is `CP` here, can you spell it out in full please